cannot co_await on just_*

```cpp
            co_await std::execution::just(42);
            co_await std::execution::just_error(std::current_exception());
            co_await std::execution::just_stopped();
```

Is there something missing to `co_await` on the `just_*` senders?  I can `co_await` on `just(42)` but not `just_error` and `just_stopped`.

```
../test/async/task.cpp:87:13: error: no member named ‘await_ready’ in ‘_P2300::execution::__just::__sender<_P2300::execution::__receivers::set_error_t, std::__exception_ptr::exception_ptr>’
   87 |             co_await std::execution::just_error(std::current_exception());
      |             ^~~~~~~~
../test/async/task.cpp:88:13: error: no member named ‘await_ready’ in ‘_P2300::execution::__just::__sender<_P2300::execution::__receivers::set_stopped_t>’
   88 |             co_await std::execution::just_stopped();
   ```

--- a/include/exec/sequence/iterate.hpp
+++ b/include/exec/sequence/iterate.hpp
@@ -163,8 +163,7 @@ namespace exec {
         template < __decays_to<__t> _Self, sequence_receiver_of<completion_signatures> _Receiver>
           requires sender_to<
             __next_sender_of_t<_Receiver, __sender_t<_Range>>,
-            __next_receiver_t<_Receiver>
-            >
+            __next_receiver_t<_Receiver> >
         friend auto tag_invoke(subscribe_t, _Self&& __self, _Receiver __rcvr) //
           noexcept(__nothrow_decay_copyable<_Receiver>)
             -> stdexec::__t<__operation<_Range, stdexec::__id<_Receiver>>> {

--- a/include/stdexec/execution.hpp
+++ b/include/stdexec/execution.hpp
@@ -1863,7 +1863,8 @@ namespace stdexec {
     };
 
     template <class _Sender, class _Promise>
-    using __value_t = __decay_t<__single_sender_value_t<_Sender, env_of_t<_Promise&>>>;
+    using __value_t = __decay_t<
+      __value_types_of_t< _Sender, env_of_t<_Promise&>, __msingle_or<void>, __msingle_or<void>>>;
 
     template <class _Sender, class _Promise>
     using __receiver_t = __t<__receiver<__id<_Promise>, __value_t<_Sender, _Promise>>>;
@@ -1924,7 +1925,8 @@ namespace stdexec {
 
     template <class _Sender, class _Promise>
     concept __awaitable_sender =
-      __single_typed_sender<_Sender, env_of_t<_Promise&>> && //
+      sender_in<_Sender, env_of_t<_Promise&>> &&             //
+      __valid<__value_t, _Sender, _Promise> &&               //
       sender_to<_Sender, __receiver_t<_Sender, _Promise>> && //
       requires(_Promise& __promise) {
         { __promise.unhandled_stopped() } -> convertible_to<__coro::coroutine_handle<>>;

--- a/test/exec/test_task.cpp
+++ b/test/exec/test_task.cpp
@@ -233,4 +233,42 @@ TEST_CASE("task - stop token is forwarded", "[types][task]") {
   CHECK(stdexec::sync_wait(scope.on_empty()));
 }
 
+TEST_CASE("task - can stop early", "[types][task]") {
+  int count = 0;
+  auto work = [&count]() -> exec::task<void> {
+    count += 1;
+    co_await [&]() -> exec::task<void> {
+      count += 2;
+      co_await stdexec::just_stopped();
+      count += 4;
+    }();
+    count += 8;
+  }();
+
+  auto res = stdexec::sync_wait(std::move(work));
+  CHECK(!res.has_value());
+  CHECK(count == 3);
+}
+
+TEST_CASE("task - can error early", "[types][task]") {
+  int count = 0;
+  auto work = [&count]() -> exec::task<void> {
+    count += 1;
+    co_await [&]() -> exec::task<void> {
+      count += 2;
+      co_await stdexec::just_error(std::runtime_error("on noes"));
+      count += 4;
+    }();
+    count += 8;
+  }();
+
+  try {
+    stdexec::sync_wait(std::move(work));
+    CHECK(false);
+  } catch (const std::runtime_error& e) {
+    CHECK(std::string_view(e.what()) == "on noes");
+  }
+  CHECK(count == 3);
+}
+
 #endif

--- a/test/exec/test_task.cpp
+++ b/test/exec/test_task.cpp
@@ -235,15 +235,15 @@ TEST_CASE("task - stop token is forwarded", "[types][task]") {
 
 TEST_CASE("task - can stop early", "[types][task]") {
   int count = 0;
-  auto work = [&count]() -> exec::task<void> {
+  auto work = [](int& count) -> exec::task<void> {
     count += 1;
-    co_await [&]() -> exec::task<void> {
+    co_await [](int& count) -> exec::task<void> {
       count += 2;
       co_await stdexec::just_stopped();
       count += 4;
-    }();
+    }(count);
     count += 8;
-  }();
+  }(count);
 
   auto res = stdexec::sync_wait(std::move(work));
   CHECK(!res.has_value());
@@ -252,15 +252,15 @@ TEST_CASE("task - can stop early", "[types][task]") {
 
 TEST_CASE("task - can error early", "[types][task]") {
   int count = 0;
-  auto work = [&count]() -> exec::task<void> {
+  auto work = [](int& count) -> exec::task<void> {
     count += 1;
-    co_await [&]() -> exec::task<void> {
+    co_await [](int& count) -> exec::task<void> {
       count += 2;
       co_await stdexec::just_error(std::runtime_error("on noes"));
       count += 4;
-    }();
+    }(count);
     count += 8;
-  }();
+  }(count);
 
   try {
     stdexec::sync_wait(std::move(work));

